### PR TITLE
feat(shuttle): Make shuttle and example-app capable of using postgres schema other than "public"

### PR DESCRIPTION
**Pull request by tybook**
## Why is this change needed?

This allows using shuttle to target e.g. multi-tenant dbs. Shouldn't be a breaking change because new parameters on public functions are optional. 

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [x] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [x] PR has a [changeset](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#35-adding-changesets)
- [ ] PR has been tagged with a change label(s) (i.e. documentation, feature, bugfix, or chore)
- [x] PR includes [documentation](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#32-writing-docs) if necessary.

---

## PR-Codex overview
This PR enables `shuttle` and its `example-app` to use a custom PostgreSQL schema. 

### Detailed summary
- Added support for custom PostgreSQL schema in shuttle and example-app
- Updated migrations to create functions with `CREATE OR REPLACE`
- Updated dependencies in `package.json`
- Added dotenv package for environment variables
- Updated README with installation instructions for node 21
- Modified database client creation to include schema parameter
- Updated table row count estimation function to include schema parameter

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/shuttle/src/example-app/app.ts [hunk 1]
```diff
@@ -40,13 +41,21 @@ const hubId = "shuttle";
...
+  private readonly dbSchema: string;
   private hubSubscriber: HubSubscriber;
   private streamConsumer: HubEventStreamConsumer;
   public redis: RedisClient;
   private readonly hubId;
 
-  constructor(db: DB, redis: RedisClient, hubSubscriber: HubSubscriber, streamConsumer: HubEventStreamConsumer) {
+  constructor(
+    db: DB,
+    dbSchema: string,
```
Comment (sanjayprabhu) [inline packages/shuttle/src/example-app/app.ts:52]: If you could add this to the end, and default to `public`, we can make it a more transparent change to the consumers

File: packages/shuttle/src/example-app/app.ts [hunk 2]
```diff
@@ -55,13 +64,14 @@ export class App implements MessageHandler {
 
   static create(
     dbUrl: string,
+    dbSchema: string,
```
Comment (sanjayprabhu) [inline packages/shuttle/src/example-app/app.ts:67]: same here

**Review by sanjayprabhu — COMMENTED** [review]
Looks good to me, would just change the function params so we don't break existing consumers.

cc @sds

File: packages/shuttle/src/example-app/app.ts (hunk 1 above)
Comment (tybook) [reply packages/shuttle/src/example-app/app.ts:52]: People shouldn't be consuming the example-app functions directly, though, no?

Comment (sds) [reply packages/shuttle/src/example-app/app.ts:52]: We should probably change this to take a configuration object so order of params doesn't matter. Would be a breaking change but we could do a minor release to signal potential incompatibilities.

Comment (sds) [reply packages/shuttle/src/example-app/app.ts:52]: (that change should happen in a separate PR—for now, let's just do what @sanjayprabhu suggested)

Comment (sanjayprabhu) [reply packages/shuttle/src/example-app/app.ts:52]: @tybook ah, you're right. I know they are using the code, but yeah upgrading this won't break them. we're probably fine. I missed that this was within the app.

**Review by sanjayprabhu — APPROVED** [review]

**Comment by tybook** [comment]
Just pushed fix for integration tests

**Review by sanjayprabhu — APPROVED** [review]

**Pull request merged**